controllers/etcdpeer_controller.go

```diff
@@ -162,6 +178,46 @@ func defineReplicaSet(peer etcdv1alpha1.EtcdPeer) appsv1.ReplicaSet {
 	}
 }
 
+func pvcForPeer(peer *etcdv1alpha1.EtcdPeer) *corev1.PersistentVolumeClaim {
+	return &corev1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      peer.Name,
+			Namespace: peer.Namespace,
+			OwnerReferences: []metav1.OwnerReference{
+				*metav1.NewControllerRef(peer, etcdv1alpha1.GroupVersion.WithKind("EtcdPeer")),
+			},
+		},
+		Spec: *peer.Spec.Storage.VolumeClaimTemplate.DeepCopy(),
+	}
+}
+
+func (r *EtcdPeerReconciler) maybeCreatePvc(ctx context.Context, peer *etcdv1alpha1.EtcdPeer) (created bool, err error) {
+	objectKey := client.ObjectKey{
+		Name:      peer.Name,
+		Namespace: peer.Namespace,
+	}
+	// Check for existing object
+	pvc := &corev1.PersistentVolumeClaim{}
+	err = r.Get(ctx, objectKey, pvc)
+	// Object exists
+	if err == nil {
+		return false, nil
+	}
+	// Error when fetching the object
+	if !apierrs.IsNotFound(err) {
+		return false, err
+	}
+	// Object does not exist
+	err = r.Create(ctx, pvcForPeer(peer))
+	if apierrs.IsAlreadyExists(err) {
+		return false, nil
```

Comment:
I think it's safest to return an error here - this implies we have a stale cache, so returning here will mean we re-reconcile once the cache has updated.